.gitignore

```diff
@@ -0,0 +1,7 @@
+src/kernels/*.db
+src/kernels/*.kdb
+
+# JetBrains IDEs
+.idea/
+cmake-build*/
+build*/
```

Comment:
My question is because that is precisely the purpose of .gitignore. Those files typically are huge in other projects. IDEs add a lot of pollution to the project tree, and we will be adding Visual Studio Code and Visual Studio IDE, too.

Reply:
Added the final version to this PR.

Reply:
> My question is because that is precisely the purpose of .gitignore.

Take a look at Linux's [.gitignore](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/.gitignore). There is nothing associated with IDEs. IDE typically creates only one directory, and this is not a big deal. You can add your favorite IDE in your own .gitignore.

Also developers usually have several build directories for convenience, so adding build directory to .gitignore makes no sense.

Reply:
FYI https://github.com/github/gitignore/tree/main

Reply:
> FYI https://github.com/github/gitignore/tree/main

Does this have anything to do with our project and will it benefit?

Reply:
I mean that any action should have a specific purpose. What is the purpose in this case?

Reply:
We prefer to keep it as is. Thank you.

Reply:
Did I understand you correctly that you refuse to discuss this?

Reply:
> I mean that any action should have a specific purpose. If there is no purpose, then it's just garbage code. What is the goal in this case?

The purpose is the `git` command to ignore project files and directories and build artifacts. So the team working on the project is more productive, and it takes less time for them to manage code changes. It also helps to automate that process, which we use in our work.

Reply:
> The purpose is the `git` command to ignore project files and directories and build artifacts. So the team working on the project is more productive, and it takes less time for them to manage code changes. It also helps to automate that process and use various tools to facilitate that, which we use in our work.

It is a bit controversial, especially part with automation. I am used to checking my changes when I commit them instead of automatically adding all changes to a new commit.

I would also like to note that these changes have nothing to do with the problem that this pull request is supposed to solve.

Reply:
> Did I understand you correctly that you refuse to discuss this?

I am not. I just have yet to notice your question. However, I prefer not to have discussions just for the sake of it. I value your time and mine as well.

Reply:
> We prefer to keep it as is in this PR. Thank you.

@junliume this kind of feedback to review comments is not acceptable. All the review comments shall be resolved before merging, one way or another.

Reply:
> > My question is because that is precisely the purpose of .gitignore.
> 
> Take a look at Linux's [.gitignore](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/.gitignore). There is nothing associated with IDEs. IDE typically creates only one directory, and this is not a big deal. You can add your favorite IDE in your own .gitignore.
> 
> Also developers usually have several build directories for convenience, so adding build directory to .gitignore makes no sense.

If i am allowed to give my opinion on this one: I think your statement doesnt make a lot of sense. The .gitignore file is to prevent git from adding files/folders to your git commits. So it makes absolute sense to add the various folders/files that the different IDEs add to the project tree get excluded, otherwise you would have to manually exclude them everytime. This is especially important if you have different developers with different IDEs like here, where every single developer would have to exclude every single file/folder from their commits manually every time. The .gitignore and the changes @apwojcik made to it are very useful imo.

The linux kernel does not neccesarily need such a precise .gitignore, as linux kernel development is done mostly by using tools like vim, which doesnt add this bulk to the project tree. Big ide´s like jetbrains IDEs or visual studio add this bulk and therefore would need to be excluded every time. I mean, you can use a local .gitignore, but why make it harder for everyone, if there could just be a single .gitignore, that correctly excludes everything?

And the issue here is, if there isnt a centrally managed .gitignore which excludes those IDEs and someone forgets to exclude e.g. the build folder, this folder is added to the commit and potentially could end up in the repo, adding bloat, making reviews harder or in the worst case might result in build errors for others.

So tl;dr:
The correct usage of a .gitignore is a very useful feature that git uses and it is definitly something that everyone could benefit from

Reply:
> > where every single developer would have to exclude every single file/folder from their commits manually every time
> 
> I agree that this file must help developers. But this does not mean that you need to add paths to folders with private photos, your diploma project files, and other files not related to the project. Each developer should manually add files to a commit, and not automatically add junk and then try to clean it up.

But why would i have to add paths to such things? These should not and probably will never end up in a project folder. But the folders that the different IDEs create do end up in the project folder. Its not like one moves them there manually and by will, the IDE just creates them there. Thats what the .gitignore is for. To get around this and exclude such folders, so the project is not bloated and "dependant" on a single IDE.

You are missing the point i guess. You dont write random paths in the .gitignore. You add paths there, that definitly could or will end up in the project folder, but are not part of the project. 

In Jetbrains IDEs cases, these folders ".idea/", "cmake-build*/", "build*/" will definitly end up in your project folder, if you use that IDE. Not matter if you want it or not (afaik you can configure the different IDEs not to do so, but thats a competely other thing and way out of this scope). So its not like you should add them to the .gitignore, you basically have to if you dont want to check for those files/folders everytime you make any changes/commits in your project. These files and folders will 100% guranteed end up in your project folder and you 100% dont want to have them in your public git repo

Reply:
> > where every single developer would have to exclude every single file/folder from their commits manually every time
> 
> I agree that this file must help developers. But this does not mean that you need to add paths to folders with private photos, your diploma project files, and other files not related to the project. Each developer should manually add files to a commit, and not automatically add junk and then try to clean it up.

I really think you are missing the point of that file, so i´ll try to explain it differently:
Lets imagine a project on github, a very simple hello world project. So you had the following (broken down) folder Structure:
- Hello_World 
   - helloworld.cpp

So you have the Hello_World project folder and a simple helloworld.cpp in that folder, nothing more. Everything fine. Now you open up that project in Visual Studion and now your project folder looks like this:
- Hello_World 
   - helloworld.cpp
   - .vs
Randomly a .vs folder appeared. Now you open the same project in a Jetbrains IDE and the folder structure looks like this:
- Hello_World 
   - helloworld.cpp
   - .vs
   - .idea
And there we got another folder, that has nothing to do with the project, but ended up in your project folder. You now make some changes to the helloworld.cpp file and want to commit, but git finds the .vs and the .idea folder and wants to add that to your commit, you would have to exclude that from your commit. You make further changes and further commits. You will propably at one point get really annyoed by the fact, that you have to exclude those 2 folders manually every single time you make a commit. Now the .gitignore comes in to play. You add those folders to your .gitignore and will never ever have to think about those folders in your git commits. Git will just skip them.

Now think about a bigger project with more than just one developer and they all use different IDEs. You dont want to rely on every developer to be able to always be 100% certain they have excluded those files and folders before they commit. Mistakes can happen and those files/folders will sooner or later end up in your git project.

To prevent this from ever happening, you exclude those files and folders, that you know will be created by the different IDEs in your .gitignore. Now no developer have to check first and be 100% certain, that they have correctly excluded every single file and folder that a commonly used IDE might have added to they project folder.

Reply:
Please pay more attention to review codes and cmakefiles. I can agree with @Joly0, but I never think to argue about ignore file is useful.

Reply:
Totally agree with @scarsty. There shouldnt need to be any argue about the ignore file. It is useful, it is a nice addition by @apwojcik and its smart he thought about adding it in this pr.
Please focus on reviewing the real code and bring this pr and project forward. Many are waiting for a final windows release to use their ROCm cards on Windows

Reply:
@Joly0 I think you should read @scarsty's comment one more time. I would also like to draw your attention to the fact that this particular pull request is a draft, it was an initial draft. It was decided to split all the work into several subsequent pull requests. If you have constructive suggestions for the entire work, you can present them by creating a new issue or pull request.